osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -234,18 +255,82 @@ private void applyOrigin(Anchor anchor)
         private Quad getSelectionQuad() =>
             GetSurroundingQuad(SelectedBlueprints.SelectMany(b => b.Item.ScreenSpaceDrawQuad.GetVertices().ToArray()));
 
-        private void applyAnchor(Anchor anchor)
+        private void applyCustomAnchor(Anchor anchor)
+        {
+            foreach (var item in SelectedItems)
+            {
+                var drawable = (Drawable)item;
+
+                item.OverridesClosestAnchor = true;
+                updateDrawableAnchor(drawable, anchor);
+            }
+        }
+
+        private void applyClosestAnchor()
         {
             foreach (var item in SelectedItems)
             {
                 var drawable = (Drawable)item;
 
-                var previousAnchor = drawable.AnchorPosition;
-                drawable.Anchor = anchor;
-                drawable.Position -= drawable.AnchorPosition - previousAnchor;
+                item.OverridesClosestAnchor = false;
+                updateDrawableAnchor(drawable, getClosestAnchorForDrawable(drawable));
             }
         }
```

Comment:
@bdach If I change these to something like this:
```cs
/// <param name="anchor">An <see cref="Anchor"/> value to apply a fixed anchor point,
/// or <see langword="null"/> if the closest point should be used.</param>
applyAnchor(ISkinnableDrawable item, Anchor? anchor)
// ...
```

Then it could be merged with `updateDrawableAnchorIfUsingClosest` as well. The loops could be moved to respective call sites. The problem is that nullable branching parameters are a bit obtuse. Thoughts?

Reply:
@bdach If these are combined, the iteration logic gets a little weird. Like I could make another method that just calls `applyClosestAnchor` to everything in `SelectedItems`, but that seems a bit weird to me?

Do you think just inlining `updateDrawableAnchorIfUsingClosest` makes more sense? It only occurs once.

```cs
public override bool HandleMovement(MoveSelectionEvent<ISkinnableDrawable> moveEvent)
{
    foreach (var c in SelectedBlueprints) // <- loop is here
    {
        Drawable drawable = (Drawable)c.Item;
        // ... and has to do this:
        drawable.Position += drawable.ScreenSpaceDeltaToParentSpace(moveEvent.ScreenSpaceDelta);

        updateDrawableAnchorIfUsingClosest(c.Item);
    }

    return true;
}

private void applyClosestAnchor()
{
    foreach (var item in SelectedItems) // <- loop again, but just does the anchor stuff
    {
        var drawable = (Drawable)item;

        item.UsesFixedAnchor = false;
        updateDrawableAnchor(drawable, getClosestAnchorForDrawable(drawable));
    }
}

private void updateDrawableAnchorIfUsingClosest(ISkinnableDrawable item)
{
    // these conditionals are unique to HandleMovement:
    if (item.UsesFixedAnchor) return;

    var drawable = (Drawable)item;

    var closestAnchor = getClosestAnchorForDrawable(drawable);

    if (closestAnchor == drawable.Anchor) return;

    // the common code is already a function:
    updateDrawableAnchor(drawable, closestAnchor);
}
```

Reply:
Actually if I just delete the `if (closestAnchor == drawable.Anchor) return;` then the code can be merged. But the looping problem is still there.